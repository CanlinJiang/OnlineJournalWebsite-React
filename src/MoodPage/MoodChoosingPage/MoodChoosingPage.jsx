import React from 'react'
import {NavLink} from "react-router-dom"
import styles from "./MoodChoosingPage.module.css"
import Frame from "/Users/littletree/Documents/GitHub/Info6150/seainfo6150-final-project-webapp/src/UtilitiesComponents/MainFrame/Frame"


export default function MoodChoosingPage() {
    const BodyComponent = () => (
        <ul className={styles.listContainer}>
            <li><NavLink to="/moodchoosing/joyful">Joyful</NavLink></li>    
            <li><NavLink to="/moodchoosing/powerful">Powerful</NavLink></li>
            <li><NavLink to="/moodchoosing/peaceful">Peaceful</NavLink></li>
            <li><NavLink to="/moodchoosing/sad">Sad</NavLink></li>
            <li><NavLink to="/moodchoosing/mad">Mad</NavLink></li>
            <li><NavLink to="/moodchoosing/scared">Scared</NavLink></li>
        </ul>
    )




    return <Frame BodyComponent={BodyComponent}/>
}
