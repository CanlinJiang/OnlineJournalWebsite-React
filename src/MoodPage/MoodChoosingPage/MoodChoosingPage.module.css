.listContainer {
    margin-top: 100px;
    margin-left: auto;
    margin-right: auto;
    margin-bottom: 100px;
    list-style-type: none;
    display: grid;
    justify-content: center;
    grid-template-rows: 300px;
    grid-template-columns: repeat(6, 200px);
    
    
    grid-auto-rows: 1fr;
    grid-gap: 30px;
}

.listContainer li{
    padding: 0;
    border: 2px solid black ;
    border-radius: 10px;
    text-align: center;
    position: relative;
}

.listContainer a{
    font-family: "prata",serif;
    width: 100%;
    height: 100%;
    text-align: center;
    
    text-decoration: none;
    color: black;
    font-size: 30px;
    
    display: flex;
    justify-content: center;
    align-items: center;

    transition: color 1s, background-color 0.5s;
}

.listContainer a:hover{
    background-color: black;
    color:white;
}

@media (max-width: 499px) {
    .listContainer {
        grid-template-rows: repeat(6, 300px);
        grid-template-columns: 200px;
    
        grid-gap: 30px;
        margin-top: 150px;
        margin-bottom: 150px;
    }
    .listContainer a{
        font-size: 24px;
    }
}

@media (min-width: 500px) and (max-width: 767px) {
    .listContainer {
        grid-template-rows: repeat(6, 400px);
        grid-template-columns: 300px;
    
        grid-gap: 30px;
        margin-top: 150px;
        margin-bottom: 150px;
    }
    .listContainer a{
        font-size: 30px;
    }
  
}

@media (min-width: 768px) and (max-width: 1199px) {
    .listContainer {
        grid-template-rows: 220px 220px;
        grid-template-columns: repeat(3, 180px);
    
        grid-gap: 30px;
        margin-top: 150px;
        margin-bottom: 150px;
    }
    .listContainer a{
        font-size: 21px;
    }
  
}

@media (min-width: 1200px) and (max-width: 1670px) {

    .listContainer {
        
        grid-template-rows: 200px;
        grid-template-columns: repeat(6, 150px);
    
        grid-gap: 20px;
        margin-top: 250px;
        margin-bottom: 300px;
    }
    .listContainer a{
        font-size: 21px;
    }
  
}

