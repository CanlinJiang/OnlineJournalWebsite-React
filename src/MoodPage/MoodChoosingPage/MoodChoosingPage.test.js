import MoodChoosingPage from "./MoodChoosingPage"
import React from "react";

describe("MoodChoosingPage tests", () => {
    it("renders correctly", () => {
        
      const { container } = render(<MoodChoosingPage />);
      expect(container).toMatchSnapshot();
    });
  
  });