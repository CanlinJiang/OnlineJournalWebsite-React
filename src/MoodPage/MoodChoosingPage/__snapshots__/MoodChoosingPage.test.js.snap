// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoodChoosingPage tests renders correctly 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="header"
    >
      <nav
        class="navigation"
      >
        <ul
          class="leftBar"
        >
          <li>
            <a
              href="/minelist"
            >
              Mine's
            </a>
          </li>
          <li>
            <a
              href="/moodchoosing"
            >
              Moods
            </a>
          </li>
          <li>
            <a
              href="/randomwalk"
            >
              Random Walk
            </a>
          </li>
        </ul>
        <ul
          class="rightBar"
        >
          <li
            id="home"
          >
            <a
              aria-current="page"
              class="active"
              href="/"
            >
              Home
            </a>
          </li>
          <li
            id="addNewArticle"
          >
            <a
              href="/addnewarticle"
            >
              Add New Article
            </a>
          </li>
        </ul>
      </nav>
    </div>
    <div
      class="body"
    >
      <ul
        class="listContainer"
      >
        <li>
          <a
            href="/moodchoosing/joyful"
          >
            Joyful
          </a>
        </li>
        <li>
          <a
            href="/moodchoosing/powerful"
          >
            Powerful
          </a>
        </li>
        <li>
          <a
            href="/moodchoosing/peaceful"
          >
            Peaceful
          </a>
        </li>
        <li>
          <a
            href="/moodchoosing/sad"
          >
            Sad
          </a>
        </li>
        <li>
          <a
            href="/moodchoosing/mad"
          >
            Mad
          </a>
        </li>
        <li>
          <a
            href="/moodchoosing/scared"
          >
            Scared
          </a>
        </li>
      </ul>
    </div>
    <div
      class="footer"
    >
      <nav>
        <ul
          class="navigation"
        >
          <li
            id="aboutMe"
          >
            <a
              href="/aboutme"
            >
              About me
            </a>
          </li>
          <li
            id="contactMe"
          >
            <a
              href="/contactme"
            >
              Contact me
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </div>
</div>
`;
